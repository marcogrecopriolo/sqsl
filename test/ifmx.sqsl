{ connect - database }
display "database sysmaster";
database sysmaster;

{ simple select }
display "simple select";
select tabid, tabname from systables where tabid <=2;

{ connect and disconnect }
display "close database";
close database;

display "connect";
connect to "sysmaster";

display "disconnect";
disconnect "sysmaster";

display "disconnect while unconnected";
whenever error continue;
disconnect "sysmaster";
display dbinfo("sqlca.sqlcode");

display "select while disconnected";
select tabid, tabname from systables where tabid <=2;
display dbinfo("sqlca.sqlcode");
whenever error stop;


display "connect using source";
connect to "sysmaster" source ifmx;

{ simple select with using }
display "simple select with using";
select tabid, tabname from systables where tabid <=? using 2;

{ simple select with format }
display "select with format";
select tabid, tabname from systables where tabid <=2 format full "[##] %s";

{ simple select with format with display labels}
display "select with format and display labels";
select tabid, tabname from systables where tabid <=2 format full "[#####] %s" headers displaylabels;

{ simple select with using and format }
display "select with using and format";
select tabid, tabname from systables where tabid <=? using 2 format full "[##] %s";

{ simple select with using format and into }
display "select with using format and into";
select tabid, tabname from systables where tabid <=? using 2 into i, n format full "[##] %s";
display i, n format full "[##] %s";

{ simple select with using format into and connection }
display "select with using format and into and connection";
select tabid, tabname from systables where tabid <=? using 2 connection "sysmaster" into i, n format full "[##] %s";
display i, n format full "[##] %s";

{ foreach }
display "foreach into";
foreach select tabid, tabname from systables where tabid <=2 into i, n
do
   display i, n format full "[##] %s";
end foreach;

{ foreach with format }
display "foreach into format";
foreach select tabid, tabname from systables where tabid <=2 into i, n
    format full "[##] %s"
do
   display i, n format full "[##] %s";
end foreach;

{ foreach with cursor }
display "foreach with cursor ";
foreach cur1 cursor for select tabid+10, tabname from systables where tabid <=2 into i, n
do
   display i, n format full "[##] %s";
end foreach;

{ foreach with cursor with hold }
display "foreach with cursor with hold";
foreach cur1 cursor with hold for select tabid+20, tabname from systables where tabid <=2 into i, n
do
   display i, n format full "[##] %s";
end foreach;

{ prepare }
display "prepare";
prepare a from select tabid, tabname from systables where tabid <=2;

{ execute }
display "execute";
execute a;

{ foreach execute }
display "foreach execute";
foreach execute a into i, n
do
   display i, n format full "[##] %s";
end foreach;

{ execute - check that foreach clauses no longer apply }
display "execute different clauses";
execute a format vertical "%i a %s";

{ execute - check that previous clauses no longer apply }
display "execute different clauses";
execute a;

{ check that we can't free or execute prepared statement being executed }
display "foreach execute - test reexecution";
whenever error continue;
foreach execute a into i, n
do
   display i, n format full "[##] %s";
   free a;
   display dbinfo("sqlca.sqlcode") format full "on free %i";
   execute a;
   display dbinfo("sqlca.sqlcode") format full "on execute %i";
end foreach;
whenever error stop;

{ free }
display "free";
free a;
whenever error continue;

display "execute freed statement";
execute a;
whenever error stop;
display dbinfo("sqlca.sqlcode");

{ prepare with placeholders }
display "prepare with placeholders";
prepare b from select tabid, tabname from systables where tabid <=?;

{ these should be done in reverse - but I seem to have a CliSDK bug
  in my hands, this should fail with a -254, not sqlnotfound }

{ execute }
display "execute w/o using";
whenever error continue;
execute b;
whenever error stop;
display dbinfo("sqlca.sqlcode") format full "clisdk defect - should be -254, get 100 instead: %i";

{ execute using }
display "execute using";
execute b using 2;

display "free again";
free b;

{ prepare with placeholders and using };
display "prepare with placeholders and using";
prepare a from select tabid, tabname from systables where tabid <=? using 2;

{ execute }
display "execute";
execute a;

{ execute using }
display "execute using";
whenever error continue;
execute a using 2;
whenever error stop;
display dbinfo("sqlca.sqlcode");

display "free again";
free a;

{ redirection }
display "create dest table";
create temp table dest(id int, name char(128));
select id, name from dest format full "[##] %s";

{ simple redirection }
display "simple redirection";
select tabid+10, tabname from systables where tabid <=2 insert into dest values(?, ?);
select id, name from dest format full "[##] %s";

{ redirection with connection clauses }
display "redirection with connection clauses";
select tabid+20, tabname from systables where tabid <=2 connection "sysmaster"
    insert into dest values(?, ?) connection "sysmaster";
select id, name from dest format full "[##] %s";

{ redirection with prepared source }
display "prepare source";
prepare c from select tabid, tabname from systables where tabid <=? using 2;
display "redirection with prepared source";
execute c insert into dest values(?, ?) connection "sysmaster";
select id, name from dest format full "[##] %s";

{ redirection with prepare source }
display "prepare target";
prepare d from insert into dest values(?, ?);
display "redirection with prepared target";
select tabid+30, tabname from systables where tabid <=2 connection "sysmaster"
    execute d;
select id, name from dest format full "[##] %s";
