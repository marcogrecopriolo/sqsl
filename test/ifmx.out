database sysmaster
simple select
1,systables
2,syscolumns
close database
connect
disconnect
disconnect while unconnected
600
select while disconnected
-349
connect using source
simple select with using
1,systables
2,syscolumns
select with format
 1 systables
 2 syscolumns
select with format and display labels
tabid tabname 
    1 systables
    2 syscolumns
select with using and format
 1 systables
 2 syscolumns
select with using format and into
 1 systables
 2 syscolumns
 2 syscolumns
select with using format and into and connection
 1 systables
 2 syscolumns
 2 syscolumns
foreach into
 1 systables
 2 syscolumns
foreach into format
 1 systables
 1 systables
 2 syscolumns
 2 syscolumns
foreach with cursor 
11 systables
12 syscolumns
foreach with cursor with hold
21 systables
22 syscolumns
prepare
execute
1,systables
2,syscolumns
foreach execute
 1 systables
 2 syscolumns
execute different clauses
 1 a 
 systables

 2 a 
 syscolumns

execute different clauses
1,systables
2,syscolumns
foreach execute - test reexecution
 1 systables
on free 620
on execute 620
 2 syscolumns
on free 620
on execute 620
free
execute freed statement
201
prepare with placeholders
execute w/o using
clisdk defect - should be -254, get 100 instead: 100
execute using
1,systables
2,syscolumns
free again
prepare with placeholders and using
execute
1,systables
2,syscolumns
execute using
622
free again
create dest table
simple redirection
11 systables
12 syscolumns
redirection with connection clauses
11 systables
12 syscolumns
21 systables
22 syscolumns
prepare source
redirection with prepared source
11 systables
12 syscolumns
21 systables
22 syscolumns
 1 systables
 2 syscolumns
prepare target
redirection with prepared target
11 systables
12 syscolumns
21 systables
22 syscolumns
 1 systables
 2 syscolumns
31 systables
32 syscolumns
